crates/sui-types/src/inner_temporary_store.rs

```diff
@@ -53,27 +53,25 @@ impl InnerTemporaryStore {
             })
             .collect();
 
-        let deleted: HashMap<_, _> = effects
+        // Add all stream-ended consensus objects to the outputkeys that then get sent to notify_commit
+        let deleted_output_keys = effects
             .deleted()
-            .iter()
+            .into_iter()
+            .chain(effects.transferred_from_consensus())
+            .chain(effects.consensus_owner_changed())
```

Comment:
should we have a single method in effects that handles both of these cases?

Reply:
I guess we could, wdyt? currently they are always used together, but I figured they are conceptually different so kept them separate

Reply:
Going to merge and work on the additional test we discussed, but pls ping if you want to follow up on this thread further